TDD_Training/TDD_Training/FizzBuzzService.cs

```diff
@@ -6,27 +6,52 @@ public class FizzBuzzService
     {
         public string CheckFizzBuzz(int givenNumber)
         {
-            switch(givenNumber)
+            try
             {
-                case int _ when IsMultipleOfThreeAndFive(givenNumber) == true:
+                if (givenNumber > 0)
                 {
-                    return "FizzBuzz";
-                }
+                    switch (givenNumber)
+                    {
+                        case int _ when IsMultipleOfThreeAndFive(givenNumber) == true:
+                        {
+                            return "FizzBuzz";
+                        }
 
-                case int x when IsMultipleOfThree(givenNumber) == true:
-                {
-                    return "Fizz";
-                }
+                        case int x when IsMultipleOfThree(givenNumber) == true:
+                        {
+                            return "Fizz";
+                        }
 
-                case int x when IsMultipleOfFive(givenNumber) == true:
-                {
-                    return "Buzz";
+                        case int x when IsMultipleOfFive(givenNumber) == true:
+                        {
+                            return "Buzz";
+                        }
+
+                        case int x when IsMultipleOfThirteen(givenNumber) == true:
+                        {
+                            return "Jazz";
+                        }
+
+                        case int x when IsMultipleOfTwentyThree(givenNumber) == true:
+                        {
+                            return "Wiz";
+                        }
+
+                        default:
+                        {
+                            return Convert.ToString(givenNumber);
+                        }
+                    }
                 }
-                default:
+                else
```

Comment:
Ideally, you should throw the exceptional cases, as soon as you can from a method and proceed with good data next.

Reply:
Fixed the comment